README.md
# victoriado

> Personal website of Victoria Do.

## [Assemble](http://assemble.io/)

Assemble is a component and static site generator that makes it dead simple to build modular sites, documentation and components from reusable templates and data.

* Documentation
* Plugins - Plugins extend the core functionality of Assemble.
* Helpers - Documentation for the helpers in the [handlebars-helpers](http://github.com/assemble/handlebars-helpers) library.

## Contributing
All contributions are welcome! The simplest way to show your support for this project is to **"star" it**. Please see [Contributing to Assemble](http://assemble.io/contributing) for more information.

## Release History
 * 2015-03-28   v0.1.0   Generated by the [Yeoman Generator](https://github.com/assemble/generator-assemble) for [Assemble](http://assemble.io)


CHANGELOG
v0.1.0:
  date: "2015-03-28"
  changes:
    - Generated by the [Yeoman Generator](https://github.com/assemble/generator-assemble) for [Assemble](http://assemble.io)


js/master.js
var home = (function() {
	var self = {};
	var $logo = $('#logo');

	self.onResize = function() {

	}

	/** 
	 * Update opacity of the logo based on scroll distance 
	 */
	self.logoController = {
		init: function() { },
		onScrollDistance: function(distance) {	
			var interaction_distance = $logo.height();
			var opacity = distance/interaction_distance;
			opacity = Math.max(1 - Math.min(1, opacity), 0.1);

			$logo.css("opacity", opacity);
		}
	}

	//Simple mechanism for changing pages with fading animation
	self.pageController = {
		init: function(opt) {
			if(!opt) return;

			var options = {
				// Required
				default: opt.default || '',
				container: opt.container || '',
				links: opt.links || '',

				// optional
				currentPage: opt.currentPage || 'works-page-current'
			};

			options.default.show('fast');
			options.default.addClass(options.currentPage);

			$(options.links).click(function(evt) {
				evt.preventDefault();

				var $target = $($(this).attr('href'));
				var $current = $('.' + options.currentPage);
				console.log($target);
				$current.fadeOut({
					duration: 'fast',
					complete: function() {
						$target.fadeIn();
					}
				});									
				//$target.fadeIn();			

				$current.removeClass(options.currentPage);
				$target.addClass(options.currentPage);

			});
		}
	}

	self.portfolioController = {
		_worksMap: {},

		init: function() {
			this.attachThumbHandler();
			this.attachThumbExpand();
			this.constructColumns($('.portfolio-gallery'), $('.portfolio-thumb'));
			this.makeWorksMap();
			this.attachLinkHandler();
			this.attachIconsHandler();
		},

		makeWorksMap: function() {
			var thumbnails = $('.portfolio-thumb');
			var links = $('.works-links li');

			if(thumbnails.length != links.length) {
				console.warn("Number of links does not match number of thumbnails.");
			}

			// returns 'name' in string formatted 'x-name'
			var extractName = function(classNames) {
				var parts = classNames.split('-');
				if(parts.length > 1) {
					return parts[1];
				}
			
				console.error("Invalid format: " + classNames);
				return null;
			}

			$.each(thumbnails, $.proxy(function(i, thumbnail) {
				thumbnail = $(thumbnail);
				var name = extractName(thumbnail.attr('id'));
				this._worksMap[name] = this._worksMap[name] || {};
				this._worksMap[name].thumbnail = thumbnail;
			}, this));
			$.each(links, $.proxy(function(i, link) {
				link = $(link);
				var name = extractName(link.attr('id'));
				this._worksMap[name] = this._worksMap[name] || {};
				this._worksMap[name].link = link;
			}, this));
		},

		attachLinkHandler: function() {
			$.each(this._worksMap, function(i, pair) {
				var $link = pair.link,
					$thumb = pair.thumbnail;

				if($thumb && $link) {
					$link.mouseenter(function() {
						$thumb.find('a').mouseover();
					}).mouseleave(function() {
						$thumb.find('a').mouseleave();
					});
				}	
			});
		},

		attachIconsHandler: function() {
			var $viewButtons = $('.open-new');

			$viewButtons.mouseenter(function() {
				$(this).find('.open-new-label').show(200, "linear");
			}).mouseleave(function() {
				$(this).find('.open-new-label').hide();
			});
		},

		attachThumbExpand: function() {
			var toggleMoreInfo = function($thumbnail) {
				$thumbnail.find('.more-info').toggle();
				$thumbnail.toggleClass('expanded');
			}

			$('.see-more-button').click(function(evt) {
				evt.preventDefault();
				var $thumbnail = $(this).parent();
				toggleMoreInfo($thumbnail);

				$thumbnail.find('.see-more').remove();
			});
		},

		attachThumbHandler: function() {
			var createSeeMore = function() {
				return $('<div class="see-more">Click to see more</div>');
			}

			var $works = $('.see-more-button');

			$works.mouseover(function(evt) {;
				evt.stopPropagation();

				var $this = $(this);
				var $parent = $this.parent();
				$this.find('.thumb-title').show();

				if(!$parent.has('.see-more').length && !$parent.is('.expanded')) {
					$this.append(createSeeMore());
				}
			}).mouseleave(function(evt) {;
				// console.log($(evt.target).find('.see-more'));
				//console.log(evt.target);
				$(this).find('.thumb-title').hide();
				$(this).find('.see-more').remove();
			});
		},

		// Create columns for the portfolio works to simulate the "masonry" style
		constructColumns: function($container, $items) {

			var itemWidth = $items.outerWidth();
			var maxWidth = $container.width();
			var n = Math.floor(maxWidth/(itemWidth + 10)); // number of cols

			var makeColumn = function(i) {
				return $('<div />')
					.width(itemWidth)
					.addClass('fl col col' + i);
			}

			var columns = [];
			for(var i = 0; i < n; i++) {
				var col = makeColumn(i);
				$container.append(col);
				columns.push(col);
			}

			$.each($items, function(i, item) {
				columns[i % n].append($(item).detach());
			});
		}		
	}


	return self;
})();


$(document).ready(function() {
	home.logoController.init();
	home.portfolioController.init();

	$(window).scroll(function() {
		home.logoController.onScrollDistance($(window).scrollTop());
	});		

	home.pageController.init({
		default: $('#page-overview'),
		container: $('#portfolio-contents'),
		links: $('.works-links a').add($('.page-back')).add($('.goto-details'))
	});

	var highlightText = function(textarea) {
		textarea.select();
	}

	$('.email-box').click(function(evt) {
		highlightText(this);
	});

	// Google Analytics
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-59103309-1', 'auto');
  ga('send', 'pageview');
	
});

Gruntfile.js
/*
 * Generated on 2015-03-28
 * generator-assemble v0.5.0
 * https://github.com/assemble/generator-assemble
 *
 * Copyright (c) 2015 Hariadi Hinta
 * Licensed under the MIT license.
 */

'use strict';

// # Globbing
// for performance reasons we're only matching one level down:
// '<%= config.src %>/templates/pages/{,*/}*.hbs'
// use this if you want to match all subfolders:
// '<%= config.src %>/templates/pages/**/*.hbs'

module.exports = function(grunt) {

  require('time-grunt')(grunt);
  require('load-grunt-tasks')(grunt);

  // Project configuration.
  grunt.initConfig({

    config: {
      src: 'src',
      dist: 'dist'
    },

    watch: {
      assemble: {
        files: ['<%= config.src %>/{content,data,templates}/{,*/}*.{md,hbs,yml}'],
        tasks: ['assemble']
      },
      livereload: {
        options: {
          livereload: '<%= connect.options.livereload %>'
        },
        files: [
          '<%= config.dist %>/{,*/}*.html',
          '<%= config.dist %>/assets/{,*/}*.css',
          '<%= config.dist %>/assets/{,*/}*.js',
          '<%= config.dist %>/assets/{,*/}*.{png,jpg,jpeg,gif,webp,svg}'
        ]
      },
      sass: {
          files: ['./src/sass/**/*.scss'],
          tasks: ['style']
      },
			scripts: {
			    files: ['./src/js/master.js'],
			    tasks: ['copy']
			 }     
    },

    connect: {
      options: {
        port: 9000,
        livereload: 35729,
        // change this to '0.0.0.0' to access the server from outside
        hostname: 'localhost'
      },
      livereload: {
        options: {
          open: true,
          base: [
            '<%= config.dist %>'
          ]
        }
      }
    },

    assemble: {
      pages: {
        options: {
          flatten: true,
          assets: '<%= config.dist %>/assets',
          layout: '<%= config.src %>/templates/layouts/default.hbs',
          data: '<%= config.src %>/data/*.{json,yml}',
          partials: '<%= config.src %>/templates/partials/*.hbs',
          plugins: ['assemble-contrib-permalinks','assemble-contrib-sitemap'],
        },
        files: {
          '<%= config.dist %>/': ['<%= config.src %>/templates/pages/*.hbs']
        }
      }
    },

    copy: {
      bootstrap: {
        expand: true,
        cwd: 'bower_components/bootstrap/dist/',
        src: '**',
        dest: '<%= config.dist %>/assets/'
      },
      theme: {
        expand: true,
        cwd: 'src/assets/',
        src: '**',
        dest: '<%= config.dist %>/assets/css/'
      },
      scripts: {
        expand: true,
        cwd: 'src/js',
        src: '**',
        dest: '<%= config.dist %>/assets/js/'
      }      
    },

    sass: {
    	dist: {
    		files: {
    			'./dist/assets/css/master.css': './src/sass/master.scss'
    		}
    	}
    },

    autoprefixer: {
      dist: {
        options: {
          browsers: ['last 2 versions', '> 1%']
        },
        files: {
          './dist/assets/css/master.css' : './dist/assets/css/master.css'
        }
      }
    },    

    cssmin: {
      add_banner: {
        options: {
        //banner: '/*! <%= pkg.name %> <%= pkg.version %> <%= grunt.template.today("dd-mm-yyyy") %> */\n',
        report: 'gzip'
        },
        files: {
          './dist/assets/css/master.min.css' : './dist/assets/css/master.css'
        }
      }
    },


    // Before generating any new files,
    // remove any previously-created files.
    clean: ['<%= config.dist %>/**/*.{html,xml}']

  });

  grunt.loadNpmTasks('assemble');
  grunt.loadNpmTasks('grunt-sass');
 	grunt.loadNpmTasks('grunt-autoprefixer');
  grunt.loadNpmTasks('grunt-contrib-cssmin'); 

  grunt.registerTask('server', [
    'build',
    'connect:livereload',
    'watch'
  ]);

  grunt.registerTask('build', [
    'clean',
    'copy',
    'assemble'
  ]);

  grunt.registerTask('default', [
    'build'
  ]);

  grunt.registerTask('style', [
  	'sass', 
  	'autoprefixer', 
  	'cssmin'
  ]);

};
